source/Nevermore/IQueryBuilder.cs

```diff
@@ -75,6 +75,13 @@ public interface IJoinSourceQueryBuilder<TRecord> : IQueryBuilder<TRecord> where
 
     public interface IQueryBuilder<TRecord> where TRecord : class
     {
+        /// <summary>
+        /// Sets the command timeout for execution of the query
+        /// </summary>
+        /// <param name="commandTimeout">A custom timeout to use for the command instead of the default.</param>
+        /// <returns>The query builder that can be used to further modify the query, or execute the query</returns>
+        IQueryBuilder<TRecord> WithTimeout(TimeSpan commandTimeout);
```

Comment:
This implementation allows you to do things like
```
transaction.Query<Customer>().InnerJoin(
    transaction.Query<People().WithTimeout(TimeSpan.FromSeconds(5))
).On(...).ToList()
```

In this case the Timeout will be ignored, which is kind of confusing.

Perhaps a better option might be to only allow you to invoke methods that execute the query after you have called `WithTimeout(...)`. So something like

```
public interface ICompleteQuery<TRecord>
{
    IEnumerable<TRecord> Stream();
    List<TRecord> ToList();
    //...all the other methods that actually executed the query go here
}
```
```
public interface IQueryBuilder<TRecord> : ICompleteQuery<TRecord>
{
    ICompleteQuery<TRecord> WithTimeout(TimeSpan commandTimeout);
}
```

Reply:
Gotcha!